styx-e2e-test/src/test/java/com/spotify/styx/e2e_tests/DatastoreCleanupTest.java

```diff
@@ -0,0 +1,79 @@
+/*-
+ * -\-\-
+ * Spotify End-to-End Integration Tests
+ * --
+ * Copyright (C) 2016 - 2019 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.e2e_tests;
+
+import static com.spotify.styx.e2e_tests.DatastoreUtil.deleteDatastoreNamespace;
+import static com.spotify.styx.e2e_tests.EndToEndTestBase.TIMESTAMP_FORMATTER;
+import static com.spotify.styx.e2e_tests.TestNamespaces.TEST_NAMESPACE_PATTERN;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.KeyQuery;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This removes old datastore test namespaces. It is not really a test.
+ */
+public class DatastoreCleanupTest {
```

Comment:
I'm a bit confused. This class together with `KubernetesCleanupTest`, when are they expected to run? Wouldn't the order matter?

Reply:
I can see in `tearDown` of base class we do try to clean up. Is this as an extra safety net? But again, this test may run in between other tests.

Reply:
Added a comment.

These cleanup classes remove test accounts/namespaces etc that might be left behind by previous e2e runs that were interrupted or teardown failed.